### DISCUSSION TITLE
Improve API for Flex component

### DISCUSSION THREAD

**Discussion by dgcollard** [CRUKorg/cruk-react-components#188] [Ideas]
can we make it more explicit like:
direction "row | column"
type "stretch | center | left | right"

_Originally posted by @Develliot in https://github.com/CRUKorg/cruk-react-components/pull/31/files/22143327057a7e7cb7bb8b4a16067007262d35f3_

**Comment by Develliot** [comment]
@dgcollard  I'm tempted to remove the component. In my current project I have made a few flex components like this: 

```
export const Row = styled.div`
  display: flex;
  flex-direction: row;
`;

export const RowStretch = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

export const RowSpaceAround = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-around;
`;

export const RowCenter = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
`;

export const ResponsiveRow = styled(Row as any)`
  display: block;
  @media (min-width: ${({
      theme: {
        breakpoint: { mobile },
      },
    }) => mobile}) {
    display: flex;
  }

....... etc.....

`;
```
I'm thinking because this is so close to the css should it even be in the component library

**Reply by dgcollard** [reply]
aye i'm inclined to agree, if we're not really providing an opinionated grid or something it might be best to leave layout concerns to the consumer app completely

**Reply by PeteLindsell** [reply]
Yeah I happy with that, there was always a question mark around what was wanted or could be useful, going up to as you said an opinionated grid component. When the CMS team start using the components this could be revisited as I think it could be more applicable to them.
